packages/core/src/components/dialog/multistepDialog.tsx

```diff
@@ -225,6 +238,21 @@ export class MultistepDialog extends AbstractPureComponent2<MultistepDialogProps
     private getDialogStepChildren(props: MultistepDialogProps & { children?: React.ReactNode } = this.props) {
         return React.Children.toArray(props.children).filter(isDialogStepElement);
     }
+
+    private getInitialIndexFromProps(props: MultistepDialogProps) {
+        if (props.initialStepIndex) {
+            const boundedInitialIndex = Math.max(
+                0,
+                Math.min(props.initialStepIndex, this.getDialogStepChildren(props).length - 1),
+            );
+            return {
+                lastViewedIndex: boundedInitialIndex,
+                selectedIndex: boundedInitialIndex,
+            };
+        } else {
+            return INITIAL_STATE;
```

Comment:
can we please inline this value here and get rid of the variable? it doesn't really make sense as this variable name anymore